### PR TITLE
M1 Mac Installation Tutorial

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
I added a detailed tutorial on how to install ZenML on M1 Macs.

Link to the gitbook page: https://app.gitbook.com/o/-MCl1Hlw9oU4xibZ-ymf/s/vlf6HVLGPRknfyMwMtR2/getting-started/installation/m1-mac-installation

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

File: docs/book/getting-started/installation/installation.md [hunk 1]
```diff
@@ -58,15 +58,6 @@ print(zenml.__version__)
 If you would like to learn more about the current release, please visit our 
 [PyPi package page.](https://pypi.org/project/zenml)
 
-## Known installation issues for M1 Mac Users
```
Comment (htahir1) [inline docs/book/getting-started/installation/installation.md:61]: I would keep this here and add a link to the bigger doc page

**Review by htahir1 — COMMENTED** [review]
Im not a Mac guy so I wouldnt know how to review this thoroughly. One thing I did notice was an absence of  "What could go wrong", to troubleshoot when known errors happen. Also I think we need a section with known limitations, e.g. not being able to use tensorflow

File: docs/book/getting-started/installation/m1-mac-installation.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+Next, we will install Homebrew under Rosetta and alias it to `brew86`.
+
+To install Homebrew under Rosetta, use the following command:
+
+```
+arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+```
+
+Then, reate the `brew86` alias like so:
```
Comment (strickvl) [inline docs/book/getting-started/installation/m1-mac-installation.md:111]:
```suggestion
Then, create the `brew86` alias like so:
```

**Review by strickvl — APPROVED** [review]
This looks great to me, barring one tiny typo. Agreed, though, on the need for a section at the bottom which lists known limitations. i.e. what can you not do under this setup...

**Comment by fa9r** [comment]
@htahir1 @strickvl what other known limitations / problems are there except for TensorFlow?

File: docs/book/getting-started/installation/m1-mac-installation.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+## 6. Installing ZenML in a Virtual Environment
+
+Lastly, let's install ZenML in a new virtual environment.
+
+### Creating a Virtual Environment with pyenv-virtualenv
+
+To create a new virtual environment called `zenml` based on Python 3.9.14, run:
+
+```
+pyenv virtualenv 3.9.14 zenml
```
Comment (schustmi) [inline docs/book/getting-started/installation/m1-mac-installation.md:187]:
```suggestion
pyenv86 virtualenv 3.9.14 zenml
```
I guess?

File: docs/book/getting-started/installation/m1-mac-installation.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+```
+
+### Activating a Virtual Environment with pyenv-virtualenv
+
+You can then activate the virtual environment using
+
+```
+pyenv activate zenml
```
Comment (schustmi) [inline docs/book/getting-started/installation/m1-mac-installation.md:195]:
```suggestion
pyenv86 activate zenml
```

**Comment by strickvl** [comment]
@fa9r no idea. That'd be more a question for @schustmi

File: docs/book/getting-started/installation/m1-mac-installation.md [hunk 5]
```diff
@@ -205,6 +205,19 @@ pip install zenml
 
 And that's it, you have successfully installed ZenML in Python 3.9 under Rosetta.
 
+## Known Limitations
+
+### TensorFlow Integration
```
Comment (strickvl) [inline docs/book/getting-started/installation/m1-mac-installation.md:210]: Do you want to mention the special Tensorflow M1 package that exists here? Or say somewhere that we're working to remove this dependency blockage. I.e. explain what options (+ tradeoffs) exist for someone who uses tensorflow but also wants to use ZenML + they have an M1/2 mac.

File: docs/book/getting-started/installation/m1-mac-installation.md (hunk 3 above)
Comment (fa9r) [reply docs/book/getting-started/installation/m1-mac-installation.md:187]: Yes! Thanks for spotting this :)

Comment (fa9r) [reply docs/book/getting-started/installation/m1-mac-installation.md:187]: Oh, no, actually you just need `pyenv86` for installation of Python versions, creating venvs and settings them active works with both versions.

File: docs/book/getting-started/installation/m1-mac-installation.md (hunk 5 above)
Comment (fa9r) [reply docs/book/getting-started/installation/m1-mac-installation.md:210]: TBH I don't know about this myself so not sure what to write here 😅

Comment (fa9r) [reply docs/book/getting-started/installation/m1-mac-installation.md:210]: I added a mention of the two MacOS TF packages and that we are working on resolving this limitation.

Comment (strickvl) [reply docs/book/getting-started/installation/m1-mac-installation.md:210]: I sent you something on Discord that maybe you can also link out to somewhere.

File: docs/book/getting-started/installation/installation.md (hunk 1 above)
Comment (fa9r) [reply docs/book/getting-started/installation/installation.md:61]: Makes sense, done.

**Review by strickvl — APPROVED** [review]
Ok this looks good to me now!

**Pull request merged**